pkg/flow/internal/controller/loader_test.go

```diff
@@ -143,6 +149,24 @@ func TestLoader(t *testing.T) {
 		require.ErrorContains(t, diags.ErrorOrNil(), `component "testcomponents.tick" must have a label`)
 	})
 
+	t.Run("Load with correct stability level", func(t *testing.T) {
+		l := controller.NewLoader(newLoaderOptionsWithStability(featuregate.StabilityBeta))
+		diags := applyFromContent(t, l, []byte(testFile), nil)
+		require.NoError(t, diags.ErrorOrNil())
+	})
+
+	t.Run("Load with below minimum stability level", func(t *testing.T) {
+		l := controller.NewLoader(newLoaderOptionsWithStability(featuregate.StabilityStable))
+		diags := applyFromContent(t, l, []byte(testFile), nil)
+		require.ErrorContains(t, diags.ErrorOrNil(), "component \"testcomponents.tick\" is at stability level \"beta\", which is below the minimum allowed stability level \"stable\"")
+	})
+
+	t.Run("Load with undefined minimum stability level", func(t *testing.T) {
+		l := controller.NewLoader(newLoaderOptionsWithStability(featuregate.StabilityUndefined))
+		diags := applyFromContent(t, l, []byte(testFile), nil)
+		require.ErrorContains(t, diags.ErrorOrNil(), "stability levels must be defined: got \"beta\" as stability of component \"testcomponents.tick\" and <invalid_stability_level> as the minimum stability level")
+	})
+
```

Comment:
Can you also add test cases for modules using a component with a disallowed stability level so we can make sure that the setting properly propagates to children controllers?